### ISSUE TITLE
Choice layer is not capable of simply forwarding input

### ISSUE THREAD

**Issue by JackTemaki** [rwth-i6/returnn#377]
I have a test case for my scenario here:
https://github.com/JackTemaki/returnn/commit/9468b301289d2df6d2b5c068babc859f0c3235e8

So if there is a network without beam-search, that provides just the feedback of any target via choice layer, the network can not be used with `task="forward"`. 

RETURNN will move layers out of the loop which are supposed to be in the loop:
```
v3: DEBUG: Rec layer 'output' (search False, train False) sub net:
v3: DEBUG:   Input layers moved out of loop: (#: 2)
v3: DEBUG:     choice
v3: DEBUG:     feedback
v3: DEBUG:   Output layers moved out of loop: (#: 1)
v3: DEBUG:     output
v3: DEBUG:   Layers in loop: (#: 6)
v3: DEBUG:     end
v3: DEBUG:     end_compare
v3: DEBUG:     stop_token_sigmoid
v3: DEBUG:     s
v3: DEBUG:     c
v3: DEBUG:     c_in
v3: DEBUG:   Unused layers: (#: 0)
v3: DEBUG:     None
```
And then fail the construction (the actual error is not relevant here)

The Problem is, that there is no way to tell a choice layer that it should simply copy the input during decoding. An alternative is to enable the search_flag, which will result in creating a beam which is wrong here, and will also lead to another crash.

There is also a comment in the code suggesting that this is not intended to work at all [here](https://github.com/rwth-i6/returnn/blob/master/returnn/tf/layers/rec.py#L4551):
```python
    else:  # no search, and no scheduled-sampling
      if not self.network.search_flag:
        assert len(self.sources) == 0  # will be filtered out in transform_config_dict
      # Note: If you want to do forwarding, without having the reference,
      # that wont work. You must do search in that case.
```

So what is the correct approach to feed the ground truth or do scheduled sampling in training, but simply copying the input during forwarding?

**Comment by patrick-wilken** [comment]
I would guess there is a way to make it work with search and beam size 1, but yes, this should not be needed.
Forwarding is simply not implemented in the layer, probably because in case of `input_type="prob"` it is not clear what forwarding means (maybe greedy search?). Also ChoiceLayer is a misnomer for regression, I guess it is not a separate layer because of the scheduled sampling implementation?

I think we have to put the handling of regression in the search code instead into a `not self.train_flag and input_type == "regression"` block and then only handle the search choices `if self.search_flag`.

**Comment by albertz** [comment]
I'm already working on it.

The intended behavior of the choice layer is to return the generated/predicted output with search flag enabled, and otherwise the ground truth.

**Comment by albertz** [comment]
Fixed via #378.
